src/singleAccount.c

```diff
@@ -0,0 +1,24 @@
+#include "state.h"
+
+static common_tx_data_t* commonData = &(instructionState.signTxContext.commonTxData);
+
+bool violatesSingleAccount(const bip44_path_t* path)
```

Comment:
As far as I understand, this both checks for violation and modifies ctx.

I am strongly in favor of separating those two things.

Reply:
I guess I'd like if ctx modifications were done in `signTx*.c`. (Is it true that we only do them there?)